File: training-data/dart/coin_change.dart
int coinChange(List<int> coins, int amount) {
  final dp = List<int>.filled(amount + 1, amount + 1);
  dp[0] = 0;
  for (var a = 1; a <= amount; a++) {
    for (final c in coins) {
      if (c <= a && dp[a - c] + 1 < dp[a]) {
        dp[a] = dp[a - c] + 1;
      }
    }
  }
  return dp[amount] > amount ? -1 : dp[amount];
}

void main() {
  print(coinChange([1, 2, 5], 11));
}

File: training-data/haskell/CoinChange.hs
import Data.Array

coinChange :: [Int] -> Int -> Int
coinChange coins amount = table ! amount
  where
    table = listArray (0, amount) [go a | a <- [0 .. amount]]
    go 0 = 0
    go a = case [1 + table ! (a - c) | c <- coins, c <= a, table ! (a - c) >= 0] of
      [] -> -1
      xs -> minimum xs

main :: IO ()
main = print (coinChange [1, 2, 5] 11)
